charters/ExpressionMatrixService/charter.md

```diff
@@ -0,0 +1,90 @@
+
+# Expression Matrix Service
+
+## Description
+
+The expression matrix service enables researchers to easily access and analyze HCA expression matrix data.
+
+## Objective
+
+* Build, operate, and maintain a data access API that serves expression
+matrices from the HCA DCP Data Store with options to subset data based on bundle metadata, genes, and numerical filters.
+
+* Serve as a test bed for a re-usable metadata query service, created by
+the *Query Service* component.
+
+## In-scope
+
+* Expression Matrix Service interface supports queries based on:
+    * Bundle identifiers
+    * Metadata (*across bundles; potentially provided directly by Query Module*)
+    * Numerical filtering (*on gene expression values, and also QC metrics*)
+    * Compound metadata and numerical filtering
+
+* Expression Service API and scalable backend that provides access to
+query interface
+
+* Architecture, software design, and scale testing for scalability
+
+* Data representation
+    * Expression matrix data format for transport
+    * Expression matrix data format for storage
+    * QC matrix data format for storage and transport
+
+
+* Extends HCA DCP Command Line Interface (CLI) client to support the Expression Matrix Service
+
+* Documentation
+    * Publish documentation for API and file formats
+    * Publish example code for developers to enable fast/easy adoption of API by analysis and visualization portals
+
+* User Experience
+    * Conduct UX research and design; drive query and API design directly from use cases
+    * Validate design with communities building portals for analysis and
+    visualization
+    * Provide feedback in the form of architecture sessions and written UX
+    reports to the DCP community about the experience of building layered services using the public API(s). (e.g., *can the expression microservice be built using only public Data Store API?*)
+    * Scientific "guardrails" to warn users when executed queries produce data subsets with questionable scientific utility
+
+* External dependencies
+    * Dependencies on Data Store, Query Service, and Secondary Analysis to enable and expose quality control metrics
+    * ~~Decision on at-rest and transfer matrix data formats~~
+    * Dependency on Query Service for join queries on metadata
+
+## Out-of-scope
+
```

Comment:
I appreciate you don't want to create things higher up the tool chain that will be developed by third parties, but to drive development it might help to identify in-scope demos or collaborations that could be used to demonstrate utility. It is important we build something that meets and demonstrates use cases, obviously.

Reply:
Perhaps identifying some of these third party clients for testing might be good?

Reply:
Thanks benedict! Yes definitely we're super excited about working with developers who want to work with our API so we can make sure we have something useful. 

We're reluctant to name anything specific in the charter though, as it's a pretty sizable expansion of scope to commit to working with particular third party tools who we don't have commensurate commitments from, and we want to remain flexible as the community's needs evolve. 🙂

Reply:
I agree that third-party demos make sense, happy to help source them through the portal group if needed. Let's not hard code demos into the charter, could put something generic outreach for adoption

Reply:
No need to commit to specific tools or demos, agree with Tim, something generic at least serves to show intent.

Reply:
Actually, I think this is covered in "Validate design with communities building portals for analysis and visualization" so is ok by me.

Reply:
@benedictpaten Do you think that the wording as identified by Tim is adequate to serve the intent you're looking for, or would you like me to add another sentence? We definitely _do_ want to make our intent to collaborate with tool builders clear.